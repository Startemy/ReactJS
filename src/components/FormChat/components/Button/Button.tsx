import React from 'react';

export const Button = () => (
  <button type="submit" data-testid="btn-chat" name='addChat'>
    Создать чат
  </button>
);
