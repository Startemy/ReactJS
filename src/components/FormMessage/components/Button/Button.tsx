import React from 'react';
import PaperAir from 'images/buttons/Paperair.png';

export const Button = () => (
  <button type="submit" data-testid="btn-send" name="send">
    <img src={PaperAir} alt="Отправить сообщение"></img>
  </button>
);
