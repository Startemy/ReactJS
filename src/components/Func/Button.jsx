import PaperAir from '../../asset/resource/images/buttons/Paperair.png'

export const Button = (props) => {
  return <button type='button'  onClick={ props.click }><img src={ PaperAir }></img></button>
}