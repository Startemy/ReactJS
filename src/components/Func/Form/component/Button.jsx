import React from 'react';
import PaperAir from '../../../../asset/resource/images/buttons/Paperair.png';

export const Button = () => {
  return (
    <button type="submit">
      <img src={PaperAir}></img>
    </button>
  );
};
