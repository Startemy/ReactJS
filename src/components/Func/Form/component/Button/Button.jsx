import React from 'react';
import PaperAir from 'images/buttons/Paperair.png';

export const Button = ({onButtonClick}) => {
  return (
    <button type="submit" data-testid="btn-send" onClick={onButtonClick}>
      <img src={PaperAir}></img>
    </button>
  );
};
