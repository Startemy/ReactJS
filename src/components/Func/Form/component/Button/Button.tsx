import React from 'react';
import PaperAir from 'images/buttons/Paperair.png';

export const Button = () => (
  <button type="submit" data-testid="btn-send">
    <img src={PaperAir} alt="Отправить сообщение"></img>
  </button>
);
