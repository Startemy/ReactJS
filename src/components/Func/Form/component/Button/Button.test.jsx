import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@storybook/testing-library';
describe('Button', () => {
  it('render component', () => {
    render(<Button />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render multiply components', () => {
    render(
      <>
        <Button />
        <Button />
      </>
    );
    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn();
    render(<Button onButtonClick={mockHandler} />);
    await userEvent.click(screen.getByTestId('btn-send'));
    expect(mockHandler).toBeCalledTimes(1);
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();
    render(<Button onButtonClick={() => setTimeout(mockHandler, 1000)} />);
    await userEvent.click(screen.getByTestId('btn-send'));
    await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
      timeout: 1100,
    });
  });
});
