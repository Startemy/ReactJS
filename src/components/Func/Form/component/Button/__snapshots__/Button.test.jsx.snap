// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="btn-send"
    type="submit"
  >
    <img
      src=""
    />
  </button>
</DocumentFragment>
`;
